public/app/features/datasources/components/picker/DataSourceDropdown.test.tsx

```diff
@@ -0,0 +1,254 @@
+  const searchBox = dropdown.container.querySelector('input');
+  expect(searchBox).toBeInTheDocument();
+  await user.click(searchBox!);
+}
+
+jest.mock('@grafana/runtime', () => {
+  const actual = jest.requireActual('@grafana/runtime');
+  return {
+    ...actual,
+    getTemplateSrv: () => {
+      return {
+        getVariables: () => [{ id: 'foo', type: 'datasource' }],
+      };
+    },
+  };
+});
+
+jest.mock('@grafana/runtime/src/services/dataSourceSrv', () => {
+  return {
+    getDataSourceSrv: () => ({
+      getList: getListMock,
+      getInstanceSettings: getInstanceSettingsMock,
+    }),
+  };
+});
+
+const pushRecentlyUsedDataSourceMock = jest.fn();
+jest.mock('../../hooks', () => {
+  const actual = jest.requireActual('../../hooks');
+  return {
+    ...actual,
+    useRecentlyUsedDataSources: () => [[xMockDS.name], pushRecentlyUsedDataSourceMock],
+  };
+});
+
+const getListMock = jest.fn();
+const getInstanceSettingsMock = jest.fn();
+beforeEach(() => {
+  getListMock.mockReturnValue(mockDSList);
+  getInstanceSettingsMock.mockReturnValue(mockDS);
+});
+
+describe('DataSourceDropdown', () => {
+  it('should render', () => {
+    expect(() => setup()).not.toThrow();
+  });
+
+  describe('configuration', () => {
+    const user = userEvent.setup();
+
+    it('should call the dataSourceSrv.getDatasourceList with the correct filters', async () => {
+      const filters = {
+        mixed: true,
+        tracing: true,
+        dashboard: true,
+        metrics: true,
+        type: 'foo',
+        annotations: true,
+        variables: true,
+        alerting: true,
+        pluginId: true,
+        logs: true,
+      };
+
+      const props = {
+        onChange: () => {},
+        current: mockDS.name,
+        ...filters,
+      };
+      window.HTMLElement.prototype.scrollIntoView = jest.fn();
+      const dropdown = render(<DataSourceDropdown {...props}></DataSourceDropdown>);
+
+      const searchBox = dropdown.container.querySelector('input');
+      expect(searchBox).toBeInTheDocument();
+      await user.click(searchBox!);
+      expect(getListMock.mock.lastCall[0]).toEqual(filters);
+    });
+
+    it('should display the current ds on top', async () => {
+      //Mock ds is set as current, it appears on top
+      getInstanceSettingsMock.mockReturnValue(mockDS);
+      await setupOpenDropdown(user, jest.fn(), mockDS.name);
+      let cards = await screen.findAllByTestId('data-source-card');
+      expect(await findByText(cards[0], mockDS.name, { selector: 'span' })).toBeInTheDocument();
+
+      //xMock ds is set as current, it appears on top
+      getInstanceSettingsMock.mockReturnValue(xMockDS);
+      await setupOpenDropdown(user, jest.fn(), xMockDS.name);
+      cards = await screen.findAllByTestId('data-source-card');
+      expect(await findByText(cards[0], xMockDS.name, { selector: 'span' })).toBeInTheDocument();
+    });
+
+    it('should get the sorting function using the correct paramters', async () => {
```

Comment:
```suggestion
    it('should get the sorting function using the correct parameters, async () => {
```

Reply:
I'm not sure about this one 🤔 

When testing the component, we're also describing the behavior. This will link the test to the implementation detail. This is fine for unit testing, but may not be that convenient for component testing.

We want to describe the proper order of the list when rendering. I'd test the order in the component renders instead of the call to the function.